### PR TITLE
feature: implement the task mgr for managing the task

### PR DESCRIPTION
**Pull request by Starnop**
### Ⅰ. Describe what this PR did
1. Implement the taskMgr
2. Add task-related routers and implement the task_bridge.

### Ⅱ. Does this pull request fix one issue?

fixes part of #366 

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

Added.

### Ⅳ. Describe how to verify it

```
# curl 127.0.0.1:8005/tasks -d '{"cID":"foo","callSystem":"foocall","dfdaemon":true,"path":"/peers/file/fooFile","rawURL":"http://aa.bb.cc.com"}'
{"ID":"00c4e7b174af7ed61c414b36ef82810ac0c98142c03e5748c00e1d1113f3c882","fileLength":-1,"pieceSize":4194304}

# curl 127.0.0.1:8005/tasks/00c4e7b174af7ed61c414b36ef82810ac0c98142c03e5748c00e1d1113f3c882
{"ID":"00c4e7b174af7ed61c414b36ef82810ac0c98142c03e5748c00e1d1113f3c882","callSystem":"foocall","cdnStatus":"WAITING","dfdaemon":true,"fileLength":-1,"pieceSize":4194304,"rawURL":"http://aa.bb.cc.com","taskURL":"http://aa.bb.cc.com"}

# curl -X PUT 127.0.0.1:8005/tasks/00c4e7b174af7ed61c414b36ef82810ac0c98142c03e5748c00e1d1113f3c882 -d '{"cdnStatus":"SUCCESS","fileLength":2000}'

# curl 127.0.0.1:8005/tasks/00c4e7b174af7ed61c414b36ef82810ac0c98142c03e5748c00e1d1113f3c882
{"ID":"00c4e7b174af7ed61c414b36ef82810ac0c98142c03e5748c00e1d1113f3c882","callSystem":"foocall","cdnStatus":"SUCCESS","dfdaemon":true,"fileLength":2000,"pieceSize":4194304,"rawURL":"http://aa.bb.cc.com","taskURL":"http://aa.bb.cc.com"}

# curl -X DELETE 127.0.0.1:8005/tasks/00c4e7b174af7ed61c414b36ef82810ac0c98142c03e5748c00e1d1113f3c882

# curl 127.0.0.1:8005/tasks/00c4e7b174af7ed61c414b36ef82810ac0c98142c03e5748c00e1d1113f3c882
{"message":"{\"Code\":502,\"Msg\":\"key: 00c4e7b174af7ed61c414b36ef82810ac0c98142c03e5748c00e1d1113f3c882: data not found\"}"}
```
### Ⅴ. Special notes for reviews
1. It hasn't been implemented for listTask yet.
2. It hasn't been implemented for `GetContentLength`.

### CODE REVIEW

File: supernode/daemon/mgr/task/manager.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+	// Step3: add a new DfgetTask
+	dfgetTask, err := tm.addDfgetTask(ctx, req, task)
+	if err != nil {
+		logrus.Infof("failed to add dfgetTask %+v: %v", dfgetTask, err)
+		return nil, err
+	}
+
+	logrus.Debugf("success to add dfgetTask %+v", dfgetTask)
+	defer func() {
```
[the remark this answers is not in the document]
Comment (Starnop) [reply supernode/daemon/mgr/task/manager.go]: Of course, we can do this using the implementation that you said.  However, we have not implement the gc module for supernode. So we should delete it in this way as a temporary solution. WDYT?

File: supernode/daemon/mgr/task/manager.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+	}
+
+	// Step2: add a new Task or update the exist task
+	task, err := addOrUpdateTask(ctx, tm.taskStore, req)
+	if err != nil {
+		logrus.Infof("failed to add or update task with req %+v: %v", req, err)
+		return nil, err
+	}
+	tm.taskLocker.getLock(task.ID, false)
+	defer tm.taskLocker.releaseLock(task.ID, false)
```
[the remark this answers is not in the document]
Comment (Starnop) [reply supernode/daemon/mgr/task/manager.go]: Updated. PTAL.

File: supernode/daemon/mgr/task/manager.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+	logrus.Debugf("success to get dfgetTask: %+v", dfgetTask)
+
+	task, err := tm.getTask(taskID)
+	if err != nil {
+		return false, nil, errors.Wrapf(err, "failed to get taskID (%s)", taskID)
+	}
+	logrus.Debugf("success to get task: %+v", task)
+
+	if dfgetTaskStatus == types.DfGetTaskStatusWAITING {
+		logrus.Debugf("start to process task start")
```
Comment (lowzj) [inline supernode/daemon/mgr/task/manager.go]: output taskId

Comment (Starnop) [reply supernode/daemon/mgr/task/manager.go]: Done.

**Comment by lowzj** [comment]
LGTM

**Pull request merged**